starport/templates/ibc/oracle/dynamic/x/{{moduleName}}/types/{{queryName}}_test.go.plush

```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	"testing"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMsg<%= queryName.UpperCamel %>Data_ValidateBasic(t *testing.T) {
+	tests := []struct {
+		name string
+		msg  Msg<%= queryName.UpperCamel %>Data
+		err  error
```

Comment:
I have a strong personal preference for only testing validity with a boolean. This generate code easier to maintain for the developer

We can also discuss it in the SPN call

Reply:
Go is a language with specifics errors we should handle one by one; in this case, I like the idea of asserting the error because we know exactly whats happens and the return, and it's clean for readability and understands the test. But it's only a matter of opinion, both are good and cover the necessary. We forget to discuss it in the call. We can do it in the next one or here.

Reply:
This is harder to maintain so I don't like to propose this convention to the developer.

Also, the content of the error message is purely for the CLI output and has no functional logic, it would be preferable to assert an error code or an error type

Reply:
I agree it's harder to maintain the messages, but we know exactly what we need to assert. My suggestion is that we assert only the error type and not the message. But the `requite.IsError` not works sometimes, like to compare similar `sdkerror` wrapped into a `spncritical`

Reply:
Do you think it can be possible to have a similar design as this one? https://github.com/tendermint/spn/pull/253
We can change the errors if needed